datadog_lambda/tracing.py

```diff
@@ -377,13 +409,125 @@ def set_dd_trace_py_root(trace_context_source, merge_xray_traces):
         )
 
 
+def create_inferred_span(event, context, function_name):
+    managed_service = detect_inferrable_span_type(event)
+    try:
+        if managed_service == ManagedService.API_GATEWAY:
+            logger.debug("API Gateway event detected. Inferring a span")
+            return create_inferred_span_from_api_gateway_event(event, context)
+        elif managed_service == ManagedService.HTTP_API:
+            logger.debug("HTTP API event detected. Inferring a span")
+            return create_inferred_span_from_http_api_event(event, context)
+        elif managed_service == ManagedService.API_GATEWAY_WEBSOCKET:
+            logger.debug("API Gateway Websocket event detected. Inferring a span")
+            return create_inferred_span_from_api_gateway_websocket_event(event, context)
+    except Exception as e:
+        logger.debug(
+            "Unable to infer span. Detected type: {}. Reason: {}", managed_service, e
+        )
+        return None
+    logger.debug("Unable to infer a span: unknown event type")
+    return None
+
+
+def detect_inferrable_span_type(event):
```

Comment:
Can you reuse (maybe some refactoring required) the same logic from https://github.com/DataDog/datadog-lambda-python/blob/main/datadog_lambda/trigger.py? Want to avoid inconsistency between what's shown by the trace and the invocation.